src/common/tusb_types.h

```diff
@@ -342,13 +343,19 @@ typedef struct TU_ATTR_PACKED
   } bmAttributes     ; ///< This field describes the endpoint's attributes when it is configured using the bConfigurationValue. \n Bits 1..0: Transfer Type \n- 00 = Control \n- 01 = Isochronous \n- 10 = Bulk \n- 11 = Interrupt \n If not an isochronous endpoint, bits 5..2 are reserved and must be set to zero. If isochronous, they are defined as follows: \n Bits 3..2: Synchronization Type \n- 00 = No Synchronization \n- 01 = Asynchronous \n- 10 = Adaptive \n- 11 = Synchronous \n Bits 5..4: Usage Type \n- 00 = Data endpoint \n- 01 = Feedback endpoint \n- 10 = Implicit feedback Data endpoint \n- 11 = Reserved \n Refer to Chapter 5 of USB 2.0 specification for more information. \n All other bits are reserved and must be reset to zero. Reserved bits must be ignored by the host.
 
   struct TU_ATTR_PACKED {
+#if defined(__CCRX__)
+    //FIXME the original defined bit field has a problem with the CCRX toolchain, so only a size field is defined
+    uint16_t size;
```

Comment:
I am curious what would be the problem with this bitfield, the bmAttributes also defined something similar but you didn't change it.

Reply:
I need to have deeper look at that. It seems to be a problem between the endian macros found in descriptor macro and the bit field definition in the `tusb_desc_endpoint_t` struct. Especially in the big endian configuration (in little endian configuration the code seems to work).
BTW: Sorry for the time delay, but at the moment I'm very busy at work. I hope to get it done this weekend

Reply:
That is problematic with endian, I would love to have some hand-on testing with this as well. I am ok to have it as it is for now.

Reply:
I did some test about this issue here. First I setup a new build variant with GCC in big endian order to check if the same problem occurs as with the CCRX. This was not successfully, build my test project with GCC in big endian order shows a  problem in the tinyUSB sources with bitfields. 

First the order of the bitfields is changed and second there is no possibility to change the ordering of the bitfields in GCC (at least I found no possibility). Every thing I tried (for example `#pragma reverse_bitfields` ) did not work.

BTW: My conclusion about this investigation is to avoid bitfields and access bit coded stuff trough macros. That maybe sounds very hard so I suggest to discuss this in a new discussion thread about how to get highly portable code. What do you think about it @hathach ?

Reply:
@Wini-Buh  unfortunately bitfield is used widely throughout the stack, it is probably too much of hassle for now to change it. I think we should still get bit endian correctly with pragma.

Reply:
@hathach Ok, if will try to investigate more on this topic. I'm not sure if it is a problem of the toolchain (my hope was to reproduce the issue with the GCC, but because it is not possible to get the same settings with the bit field ordering the try was hopeless).